src/Cloud/weglide/UploadFlight.cpp

```diff
@@ -58,27 +59,43 @@ MakeUploadFlightMime(CURL *easy, const WeGlideSettings &settings,
 }
 
 Co::Task<boost::json::value>
-UploadFlight(CurlGlobal &curl, const WeGlideSettings &settings,
+UploadFlight(CurlGlobal& curl, const char *url,
+             const struct Pilot &pilot,
              uint_least32_t glider_type,
              Path igc_path,
              ProgressListener &progress)
 {
-  NarrowString<0x200> url(settings.default_url);
-  url += "/igcfile";
-
   CurlEasy easy{url};
   Curl::Setup(easy);
   const Net::ProgressAdapter progress_adapter{easy, progress};
-  easy.SetFailOnError();
+  // easy.SetFailOnError disabled: HTTP errors are dealt with here at the end
 
-  const auto mime = MakeUploadFlightMime(easy.Get(), settings,
+  const auto mime = MakeUploadFlightMime(easy.Get(), pilot,
                                          glider_type, igc_path);
   easy.SetMimePost(mime.get());
 
   Json::ParserOutputStream parser;
   const auto response =
     co_await Curl::CoStreamRequest(curl, std::move(easy), parser);
-  co_return parser.Finish();
+  auto json =  parser.Finish();
+  std::string error_msg = "";
+
+  if (response.status < 400) {  // 4xx and 5xx are fail codes
+    co_return json;  // return with the WeGlide upload json object
+  } else {
+    // a http error occurs, but not in curl - no SetFailOnError()
+    switch (response.status) {
+    case 400: error_msg = "HTTP failure code: 400 Bad Request:\n";
+      error_msg += "Probably one of the parameter is wrong!";
+      break;
+    case 406: error_msg = "HTTP failure code: 406 Not Acceptable:\n";
+      error_msg += "The IGC file probably already exists on the server!";
+      break;
+    default: error_msg = "HTTP failure code: "
+      + std::to_string(response.status); break;
```

Comment:
`std::to_string()` adds a lot of overhead (both code bloat and heap allocations); better use `FormatRuntimeError()`.